aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -181,13 +181,34 @@ def export_tree(
             )
 
     # Add all the nodes contained within the specified groups
-    for group in given_groups:
-        for entry in group.nodes:
-            entities_starting_set[NODE_ENTITY_NAME].add(entry.uuid)
-            if issubclass(entry.__class__, orm.Data):
-                given_data_entry_ids.add(entry.pk)
-            elif issubclass(entry.__class__, orm.ProcessNode):
-                given_calculation_entry_ids.add(entry.pk)
+    if given_group_entry_ids:
+
+        if not silent:
+            print('RETRIEVING NODES FROM GROUPS...')
+
+        # Use single query instead of given_group.nodes iterator for performance.
+        qh_groups = orm.QueryBuilder().append(
+            orm.Group, filters={
+                'id': {
+                    'in': given_group_entry_ids
+                }
+            }, tag='groups'
+        ).queryhelp
+
+        data_results = orm.QueryBuilder(**qh_groups).append(orm.Data, project=['id', 'uuid'], with_group='groups').all()
+
+        from builtins import zip  # pylint: disable=redefined-builtin
```

Comment:
Without this I get an error that `zip` is a module and not callable ( we have a module called `zip` in the same folder).
If you want me to rename the module instead, let me know.